server/src/main/java/org/elasticsearch/cluster/metadata/SystemIndexMetadataUpgradeService.java

```diff
@@ -81,6 +86,13 @@ public ClusterState execute(ClusterState currentState) throws Exception {
                     if (isSystem != cursor.value.isSystem()) {
                         updatedMetadata.add(IndexMetadata.builder(cursor.value).system(cursor.value.isSystem() == false).build());
                     }
+
+                    if (isSystem) {
+                        if (cursor.value.getSettings().getAsBoolean(IndexMetadata.SETTING_INDEX_HIDDEN, false)) {
+                            throw new IllegalStateException("Cannot define index [" + cursor.value.getIndex().getName() +
```

Comment:
I don't believe this should throw an error but rather ensure that the setting is set to false and if not, correct it, in this task.

This validation should happen elsewhere. Maybe in the `MetadataUpdateSettingsService` or the setting validator?

Reply:
In `TransportUpdateSettingsAction`, we validate that a settings update with unknown origin isn't trying to change the settings of a system index. Since that validation already brings together cluster state, system indices, and index settings, it seemed like a good place to add another layer of validation.